### DISCUSSION TITLE
WSL installation not working.

### DISCUSSION THREAD

**Discussion by SaikatBaidya** [Cyfrin/foundry-full-course-cu#2263] [Q&A]
I have Windows 10 build 19045. I have tried running the wsl --install, wsl.exe --install --no-distribution, etc. It installs but when I reboot, the distro does not open up (I have ubuntu installed). When I run the wsl -l -v, it gives the following error:

` ` `
This application requires the Windows Subsystem for Linux Optional Component.
Install it by running: wsl.exe --install --no-distribution
The system may need to be restarted so the changes can take effect.
Error code: Wsl/WSL_E_WSL_OPTIONAL_COMPONENT_REQUIRED
` ` `

Kindly help!!!

**Comment by EngrPips** [comment]
Hello @cromewar, we need your help here sir.

**Reply by SaikatBaidya** [reply]
Hey, so instead of Ubuntu, I tried WSL with Kali-Linux and it's working fine. Can I use Kali linux for the rest of the course?

**Reply by EngrPips** [reply]
If it is working fine for you, yeah sure use it.
